lib/plotting_functions.py

```diff
@@ -172,6 +183,44 @@ def global_average(fld, wgt, verbose=False):
     return np.ma.average(avg1)
 
 
+def spatial_average(indata, weights=None, spatial_dims=None):
+    import warnings
+
+    if weights is None:
+        #Calculate spatial weights:
+        if 'lat' in indata.coords:
+            weights = np.cos(np.deg2rad(indata.lat))
+            weights.name = "weights"
+        elif 'ncol' in indata.dims:
+            if 'area' in indata:
+                warnings.warn("area variable being used to generated normalized weights.")
+                weights = indata['area'] / indata['area'].sum()
+            else:
+                warnings.warn("We need a way to get area variable. Using equal weights.")
+                weights = xr.DataArray(1.)
+            weights.name = "weights"
+        else:
+            weights = xr.DataArray(1.)
+            weights.name = "weights"
+        #End if
+    #End if
+
+    #Apply weights to input data:
+    weighted = indata.weighted(weights)
+
+    # we want to average over all non-time dimensions
+    if spatial_dims is None:
+        if 'ncol' in indata.dims:
+            spatial_dims = ['ncol']
+        else:
+            spatial_dims = [dimname for dimname in indata.dims if (('lat' in dimname.lower()) or ('lon' in dimname.lower()))]
+
+    if not spatial_dims:
+        warnings.warn("No spatial dimensions were identified, so can not perform average.")
+
+    return weighted.mean(dim=spatial_dims)
```

Comment:
Looks like `arr.mean(dim=[])` just returns the original `arr` without doing averaging.

Reply:
Thanks for looking that up @brianpm!  I am worried that if we return the `weighted` array without any dimension reductions from averaging that the scripts using this function could produce un-expected or un-predictable behavior that might be really difficult to track down (e.g. everything "works" but the final values are wrong).  Given this I might go ahead and make the `if not spatial_dims` check and error instead of a warning.  If you or @justin-richling disagree with that plan please let me know.  Thanks!

Reply:
That is perfectly ok with me.